distribution/pull_v2.go

```diff
@@ -52,6 +53,17 @@ func (e imageConfigPullError) Error() string {
 	return "error pulling image configuration: " + e.Err.Error()
 }
 
+// unsupportedMediaTypeError is an error issued when attempted
+// to pull unsupported content.
+type unsupportedMediaTypeError struct {
+	MediaType string
+}
+
+// Error returns the error string for unsupportedMediaTypeError.
+func (e unsupportedMediaTypeError) Error() string {
```

Comment:
This should probably implement errdefs.InvalidParameter, to prevent returning a 5xx status through the API (haven't checked what status code it currently returns though)

Reply:
Status code is 200 in the current case (before this change), and this change doesn't affect it.
Logic behind this behaviour is strange to me: https://github.com/moby/moby/blob/542c735926c9208dc0fe9fc02026a110ca0343f9/api/server/router/image/image_routes.go#L74
If the error has been flushed to the output stream, then we don't return an error from the handler.

Reply:
Ah, good point; forgot this was a steaming endpoint.

Reply:
Thanks, done!